include/flatbuffers/idl.h

```diff
@@ -558,6 +561,7 @@ struct IDLOptions {
   std::string cpp_object_api_pointer_type;
   std::string cpp_object_api_string_type;
   bool cpp_object_api_string_flexible_constructor;
+  int cpp_object_api_field_case;
```

Comment:
I think this option might be C++-specific like the `--cpp-std` option. It is unnecessary to add it into the `IDLOptions`.

Reply:
all C++ codegen options are in that place, this is together with the others

Reply:
I know about it.
In my opinion, `IDLOptions` should contain only common options connected with the parser or with the IDL-generator (bfbs). All other options better to hide in specific code generators (e.g. `IDLOptionsCpp`).

@aardappel what do think about this?

Reply:
@vglavnyy @aardappel there are 5 or 6 cpp options in IDLOptions, only 2 in IDLOptionsCpp; I suggest a followup PR to migrate and consolidate all the C++ codegen options into IDLOptionsCpp, which I'm happy to contribute once this is merged